catroid/src/org/catrobat/catroid/ui/adapter/LookAdapter.java

```diff
@@ -59,7 +60,14 @@ public long getItemId(int position) {
 			return INVALID_ID;
 		}
 		LookData item = getItem(position);
-		return idMap.get(item);
+
+		for (Map.Entry<LookData, Integer> e : idMap.entrySet()) {
+			if (item.equals(e.getKey())) {
+				return e.getValue();
+			}
+		}
+
+		return INVALID_ID;
```

Comment:
Ah, that's the problem. Thanks for the update! :smile_cat:
I created a ticket for the hashCode. https://jira.catrob.at/browse/CAT-2068